onnxruntime/core/graph/model.cc

```diff
@@ -98,6 +99,23 @@ Model::Model(const std::string& graph_name,
   GSL_SUPPRESS(r .11)
   graph_.reset(new Graph(*this, model_proto_.mutable_graph(), *p_domain_to_version, IrVersion(), schema_registry,
                          model_functions, logger));
+
+  std::vector<FunctionTemplate*> model_local_func_templates;
+  for (auto& func : model_local_functions) {
+    auto func_schema_ptr = function_utils::CreateSchema(func.domain(), 
+        func.name(), 
+        graph_->GetModelLocalFunctions(), 
+        graph_->DomainToVersionMap(),
+        *schema_registry,
+        logger,
+        allow_released_opsets_only);
+    auto func_template_ptr = std::make_unique<FunctionTemplate>();
+    func_template_ptr->op_schema_ = std::move(func_schema_ptr);
+    func_template_ptr->onnx_func_proto_ = &func;
+    model_local_function_templates_.push_back(std::move(func_template_ptr));
```

Comment:
I think this is a programming error.  In line 92 to line 96, it made a copy of "func". But here you didn't use the copy. You directly take the pointer of the original parameter. Which means the life time of this model object is bounded to the input parameter.